src/server/game/Spells/Spell.cpp

```diff
@@ -2495,8 +2495,8 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
         // Needs to be called after dealing damage/healing to not remove breaking on damage auras
         DoTriggersOnSpellHit(spellHitTarget, mask);
 
-        // if target is fallged for pvp also flag caster if a player
-        if (unit->IsPvP() && m_caster->GetTypeId() == TYPEID_PLAYER)
+        // if it's not selfcast and target is flagged for pvp also flag caster if a player
+        if (spellHitTarget != m_caster && spellHitTarget->IsPvP() && m_caster->GetTypeId() == TYPEID_PLAYER)
```

Comment:
This is wrong. `spellHitTarget` points back to m_caster if spell was reflected (but it should still flag for PvP), using `unit` was correct here.